File: MasterLogicaProgramacion/EjerciciosPython/Ej15.py
def caracterMasUsado(texto, incluirEspacios = False):
    texto = texto.lower()
    caracteres = []
    [caracteres.append(x) for x in texto if x not in caracteres]

    if not incluirEspacios:
        caracteres.remove(" ")

    caracterMasUsado = ""
    masRepetido = 0

    for c in caracteres:
        repeticiones = len([a for a in texto if a == c])

        if repeticiones > masRepetido:
            masRepetido = repeticiones
            caracterMasUsado = c

    if caracterMasUsado == " ":
        caracterMasUsado = "[espacio]"
    
    print("Caracter Mas Usado: " + caracterMasUsado + ", " + str(masRepetido) + " veces")
    return caracterMasUsado

print(caracterMasUsado("OLA KE ASE PROGRAMANDO O KE ASE", False))

File: Cursos/MasterLogicaProgramacion/EjerciciosJavascript/Ej15.js
function caracterMasUsado(texto, incluirEspacios = false){
    texto = texto.toLowerCase();
    let caracteres = texto.split("").filter((item, index) => texto.indexOf(item) === index);

    if(!incluirEspacios){
        caracteres = caracteres.filter(x => x != " ");
    }

    let caracterMasUsado = "";
    let masRepetido = 0;

    for (c of caracteres) {
        let repeticiones = (texto.split("").filter(x => x == c)).length;

        if (repeticiones > masRepetido) {
            masRepetido = repeticiones;
            caracterMasUsado = c;
        }
    }

    if(caracterMasUsado == " ") caracterMasUsado = "[espacio]";
    
    console.log("Caracter Mas Usado: " + caracterMasUsado + ", " + masRepetido + " veces");
    return caracterMasUsado;
}

console.log(caracterMasUsado("OLA KE ASE PROGRAMANDO O KE ASE", true));